### ISSUE TITLE
Re-work output of available resources and indicators to be more legible

### ISSUE THREAD

**Issue by karpfen** [mapme-initiative/mapme.biodiversity#236]
# Background

With increasing numbers of resources and indicators it becomes harder for users to keep an overview of available indicators.

In light of #232, it would also make sense to align resource and indicator names. I think in general it is nice to stick to the convention of using a source name (e.g. the data providing organization) followed by the indicator (e.g. `gfw_treecover`) wherever possible.

One thing to consider, though: users may not know what the acronyms gfw, gsw, gmw, ... stand for and need a convenient way to figure it out. Which is why I suggest we provide a...

# Clear, readable table of available resources and indicators

`available_indicators()` and `available_resources()` do their job, but I'd like to have something more legible in a nice tabular form.
I'm imagining something similar to this:

```
library(mapme.biodiversity)

ind <- available_indicators()

res <- as.character(sapply(ind, function(x) {paste0(names(x$resources), collapse = ", ")}))
pm <- as.character(sapply(ind, function(x) {paste0(names(x$processing_mode), collapse = ", ")}))
arg <- as.character(sapply(ind, function(x) {paste0(names(x$arguments), collapse = ", ")}))

tibble(
  names(ind), res, pm, arg
)
```

Maybe with one additional column that's a bit more descriptive than just the abbreviated indicator names.

# To dos

Based on the points above, I suggest these action points. Anything is up for discussion of course.

- [ ] Align indicator and resource names
- [ ] Apply (wherever possible) to naming conventions
- [ ] Add overview table

**Comment by goergen95** [comment]
Great proposal and I am absolutely supporting this! One thing to keep in mind is that if we implemented #232 there is actually no longer a technical need for the register functions, since the arguments as well as the resource/indicator functions are provided to the backend during user run-time. I think this actually benefits your proposal, because we could still keep the register functionality, however, now with the single purpose to produce a nice overview for users, e.g. by default returning a tibble with yet to be determined variables etc. Also, experienced users aiming at extending the backend with their own resources/indicators no longer are required to use the register functions if they have no need for it.

**Comment by goergen95** [comment]
hi @karpfen, wanted to revive this as I am currently re-working the register functionality as mentioned in my last comment. Would appreciate your input on what you think would be valuable information to convey to users of the package via `available_*()`. I would require your feedback before the end of the month, as this is the deadline to announce the dev branch available for testing.

**Comment by karpfen** [comment]
Hi @goergen95, in general, I have relatively simple tables in mind that don't exhaustively contain all information on all indicators but rather give a more compact overview about what's possible with the package.
Right now, the `available_*` functions contain all the information, but you kind of have to know you have to use something like `names(available_indicators())` and then continue with `available_indicators()$indicatorXYZ` from there.

I think a solution that lists the resources/indicators in just a single table and the user can then dig deeper with `?indicatorXYZ` would make these resources more ~~accessible~~ discoverable in the end.


For _how_ to achieve this, I see two options.

1. "Auto generated": This would eliminate the need for the `register_*` calls.
- `available_resources()` generates a table listing the 
  - Title text of the reference document (if that is possible)
  - `@name`,
  - `@format`,
  - (`@reference`,)
  - (`@source`)

of all documentation with `@docType == 'data'` and `@keywords == 'resource'`.

- `available_indicators()` generates a table listing the name of the
  - Title or description text of the reference document (if that is possible)
  - `@name` of the related resource if somehow possible
  - `@name` of the indicator

of all documentation with`@docType == 'data'` and `@keywords == 'indicator'`.

2. Using information from `register_indicator()`: Here I'm basically imagining the approach from the initial post. It just lists the contents that have been passed to the `register_*` functions in a table rather than a list.
I'm imagining that this is technically easier, although it would keep the burden of filling the `register_*` functions on the developer.

What do you think, does that make sense?

**Comment by goergen95** [comment]
Thanks, a version is now up on `dev` at revision e559e098. I opted for the non-automated way for now. `available_resources()` returns a flat tibble with columns name, type, source, and license. `available_indicators()` now returns a tibble with columns name, description, and resources, where resources is a nested-list column with the required resources as outputted from `available_resources()`.

**Comment by goergen95** [comment]
@karpfen: another question - is it within the scope of this issus to rename resources/indicators?

**Comment by karpfen** [comment]
I'm all for it. I think this is a good opportunity to make the naming a bit more stringent.

**Comment by goergen95** [comment]
Agreed, I also would like to reconsider how we name resources/indicators. I will post a proposal in this issue later this week and we can give others some time to respond. I could imagine to pull up a first draft towards end of April and merge it into main with the second milestone towards end of May (see #240).

**Comment by goergen95** [comment]
Just reconsidered: I think its best to limit the scope of this issue to the already implemented re-work of the available_*() functions and open a new issue for renaming resources/indicators.

**Comment by goergen95** [comment]
after discussion with @Jo-Schie this could be further improved for discoverability for new users:

- add description column for resources
- add rendered tables for resources/indicators to readme

**Comment by goergen95** [comment]
@Jo-Schie and @karpfen: please see the GitHub rendered version if the [README](https://github.com/mapme-initiative/mapme.biodiversity/tree/dev) and leave some feedback?

**Comment by karpfen** [comment]
Cool, this is much more legible than before.
One point regarding the code example: The code is wrapped in parentheses (starting at https://github.com/mapme-initiative/mapme.biodiversity/blob/dev/README.Rmd#L111), this only saves us a `print()` statement, right?
I'd actually remove them, just because it may look unfamiliar to a lot of people.

Maybe the printing could even be done in a separate chunk with `echo=FALSE`, then it looks even cleaner. What do you think, @goergen95 ?

**Comment by goergen95** [comment]
Released with version v0.6.0
